core/src/main/java/io/opencensus/stats/NoopStats.java

```diff
@@ -99,9 +110,28 @@ public void setState(StatsCollectionState state) {
     static final StatsRecorder INSTANCE = new NoopStatsRecorder();
 
     @Override
-    public void record(TagContext tags, MeasureMap measureValues) {
+    public StatsRecord newRecord() {
+      return getNoopStatsRecord();
+    }
+  }
+
+  @Immutable
+  private static final class NoopStatsRecord extends StatsRecord {
+    static final StatsRecord INSTANCE = new NoopStatsRecord();
+
+    @Override
+    public StatsRecord put(MeasureDouble measure, double value) {
+      return this;
+    }
+
+    @Override
+    public StatsRecord put(MeasureLong measure, long value) {
+      return this;
+    }
+
+    @Override
+    public void recordWithExplicitTagContext(TagContext tags) {
```

Comment:
Probably at one point we should add another API that is:

public void recordWithExplicitTagContextAndTags(TagContext tags, Map<String, String> extraTags) {
}

To allow users to record stats with extra tags without creating a new TagContext only for this call.

Not necessary now but something to keep in mind.